faucet/src/faucet.rs

```diff
@@ -343,10 +344,15 @@ pub fn run_local_faucet_with_port(
     });
 }
 
-// For integration tests. Listens on random open port and reports port to Sender.
+/// For integration tests. Listens on a random open port and reports the port to Sender.
+/// If the NEXTEST environment variable is set, a non-overlapping port is allocated instead.
 pub fn run_local_faucet(faucet_keypair: Keypair, per_time_cap: Option<u64>) -> SocketAddr {
+    let port: u16 = match std::env::var("NEXTEST") {
```

Comment:
I think a better gate to use is debug_assertions. Using NEXTEST here will still allow this to break if running under plain cargo test.